src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
Seems doable, you could potentially even calculate several splits at once (take the top 3 principle components). One thing to keep in mind with PCA though is that it's susceptible to outliers so you aren't really guaranteed an even split, especially if your sample isn't very gaussian-like. In that case, to maximize the variance it may find a split that overemphasizes 1 or 2 outliers in your sampled set.